### ISSUE TITLE
bgverify fails to compile with clang

### ISSUE THREAD

**Issue by ant1** [cmu-sei/BigGrep#1]
bgverify fails to compile with clang

The failure is:

```
In file included from bgverify.cpp:22:
./Logger.hpp:457:5: error: incompatible operand types ('int' and '__ostream_type' (aka 'basic_ostream<char, std::char_traits<char> >'))
    LOG(lvl) << enter_msg << std::endl;
    ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
./Logger.hpp:398:82: note: expanded from macro 'LOG'
#define LOG(level) (level < Logger::logger()->getLoggerLevel(__FILE__,__LINE__)) ? LNOOP : Logger::logger()->log(level)
                   ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ ^
```

**Comment by ant1** [comment]
It builds fine with patch below:

```
diff --git a/src/Logger.hpp b/src/Logger.hpp
index 3bf7fd7..2a167b0 100755
--- a/src/Logger.hpp
+++ b/src/Logger.hpp
@@ -362,7 +362,7 @@ private:
 };
 
 // macro version, usage like so: LOG(Logger::WARNING) << "Hm...something looks suspicious";
-#if 0
+#if 1
 #define LOG(level) \
   if (level < Logger::logger()->getLoggerLevel()) ; \
   else Logger::logger()->log(level)
```

**Comment by sei-chines** [comment]
In a slightly more updated version of that logging code that I have in another (internal) project I actually changed the macros to use short circuit evaluation instead of the ternary operator (or the older if/else style you switched to), like so:
```

#define LOG(level) (level >= Logger::logger()->getLoggerLevel(__FILE__,__LINE__)) && Logger::logger()->log(level) 
#define LOGcnt(level) (level >= Logger::logger()->getLoggerLevel(__FILE__,__LINE__)) && Logger::logger()->log(level,false) 
```

Does that work for you under clang by any chance?

**Comment by ant1** [comment]
Yes this builds fine with clang.

**Comment by sei-chines** [comment]
Cool, I'll let the person who took over this code know then and see about getting that fix pulled in.

**Comment by ant1** [comment]
Fixed in commit e461cdf
